src/NHSD.BuyingCatalogue.Ordering.Api/Controllers/OrderingPartyController.cs

```diff
@@ -74,18 +76,61 @@ public async Task<ActionResult<OrderingPartyModel>> GetAsync(string orderId)
         }
 
         [HttpPut]
-        public ActionResult Update(string orderId, OrderingPartyModel model)
+        public async Task<ActionResult> UpdateAsync(string orderId, OrderingPartyModel model)
         {
-            if (orderId is null)
+            if (model is null)
+            {
+                throw new ArgumentNullException(nameof(model));
+            }
+
+            var order = await _orderRepository.GetOrderByIdAsync(orderId);
+            if (order is null)
             {
                 return NotFound();
             }
 
-            if (model is null)
+            var primaryOrganisationId = User.GetPrimaryOrganisationId();
+            if (primaryOrganisationId != order.OrganisationId)
             {
-                throw new ArgumentNullException(nameof(model));
+                return Forbid();
             }
 
+            var primaryContact = model.PrimaryContact;
+
+            var contact = new Contact
+            {
+                FirstName = primaryContact.FirstName,
+                LastName = primaryContact.LastName,
+                Email = primaryContact.EmailAddress,
+                Phone = primaryContact.TelephoneNumber
+            };
+
+            order.OrganisationContact = contact;
+
+            var organisationModel = model.Organisation;
+            var addressModel = organisationModel.Address;
+            var address = new Address

```

Comment:
I've actually just pushed an extension for the address, also you dont want to new it up everytime you update, or it will create another address. The extension fixes that problem though. Its the same for contact